pkg/kademlia/routing.go

```diff
@@ -321,3 +322,77 @@ func (rt *RoutingTable) ConnSuccess(ctx context.Context, node *pb.Node) {
 		zap.L().Debug("connection success error:", zap.Error(err))
 	}
 }
+
+func splitBucket(bIDs []bucketID, bitDepth int) ([]bucketID, []bucketID) {
+	var b0 []bucketID
+	var b1 []bucketID
+
+	for _, bID := range bIDs {
+		byteDepth := bitDepth / 8
+		bitOffset := bitDepth % 8
+		power := uint(7 - bitOffset)
+		bitMask := byte(1 << power)
+		b := bID[byteDepth]
+		if b&bitMask > 0 {
+			b1 = append(b1, bID)
+		} else {
+			b0 = append(b0, bID)
+		}
+	}
+	return b0, b1
+}
+
+// BufferedGraph prints the routing table graph as a dot graph in the specified buffer
+func (rt *RoutingTable) BufferedGraph(buf *bytes.Buffer) {
+	buf.Write([]byte("digraph{\nnode [shape=box];edge [dir=none];\n"))
+
+	ids, _ := rt.GetBucketIds()
+	var bucketids []bucketID
+	for _, n := range ids {
+		bucketids = append(bucketids, keyToBucketID(n))
+	}
+
+	rt.addBucketsToGraph(bucketids, 0, buf, "")
+
+	buf.Write([]byte("}\n"))
```

Comment:
Add this above as `defer buf.Write([]byte("}\n"))`, it helps to avoid accidentally forgetting the closing brace.